### DISCUSSION TITLE
Automatic deploy with Github webhook error

### DISCUSSION THREAD

**Discussion by Vanja-S** [Dokploy/dokploy#142] [Q&A] [answered]
Hi team, I set up a Github webhook on release create event, however the dokploy api responds with a 301 and does not deploy the application.

Here is the response and error body:

**Headers**
```
Content-Length: 30
Content-Type: application/json; charset=utf-8
Date: Thu, 13 Jun 2024 19:47:44 GMT
Etag: "f26sj40jkiu"
Vary: Accept-Encoding
```

**Body**
```
{"message":"Branch Not Match"}
```

My question is which branches do not match? I have my config on dokploy set to deploy from the main branch (root path) from which I also create releases. Any ideas?

**Comment by Siumauricio** [comment]
Simple because the branch you selected in your application will only trigger the webhook there, when you make a push for example from another branch it will not trigger only when you make changes specifically to the branch you selected in the dokploy configuration.

in short, if you selected the main branch in dokploy, only when you push that branch the webhook will trigger, otherwise you will get that.

**Reply by Vanja-S** [reply]
So maybe I am missing some knowledge of GitHub internals, but creating a new release on main also creates a new tag. However, the payload the webhook sends includes a lot of info, and in between all that you find:

```json
"release" : {
    // ...
    "target_commitish": "main",
    // ...
}
```

which points to the main branch, maybe dokploy is looking at some other value in the whole JSON?

**Reply by Siumauricio** [reply, marked as answer]
No, we only trigger the deployment when you push on a branch that you've selected in dokploy

https://github.com/Dokploy/dokploy/blob/canary/pages/api/deploy/%5BrefreshToken%5D.ts#L56

**Reply by Vanja-S** [reply]
Got it, thanks for the answer!

Although maybe supporting other custom triggers could be an option?

**Reply by Paulitos** [reply]
> No, we only trigger the deployment when you push on a branch that you've selected in dokploy
> 
> https://github.com/Dokploy/dokploy/blob/canary/pages/api/deploy/%5BrefreshToken%5D.ts#L56

The link no longer works.
